src/services/account/useAccountDetails.ts

```diff
@@ -0,0 +1,193 @@
+import { useQuery } from '@tanstack/react-query'
+import { z } from 'zod'
+
+import Api from 'shared/api'
+import { NetworkErrorObject } from 'shared/api/helpers'
+
+export const InvoiceSchema = z
+  .object({
+    amountDue: z.number().nullable(),
+    amountPaid: z.number().nullable(),
+    amountRemaining: z.number().nullable(),
+    currency: z.string().nullable(),
+    customerAddress: z.object({}).nullish(),
+    customerName: z.string().nullable(),
```

Comment:
JC how did you verify when to use nullable vs not?

Reply:
I honestly used the existing mock [here](https://github.com/codecov/gazebo/blob/0e05dd97a9a06d50facc3045ad914ce616a10a1d/src/services/account/mocks.js#L13) as a guide. It was a bit hard to parse from codecov-api, but I _think_ this was pretty close to actual.

API model: https://github.com/codecov/codecov-api/blob/daacd21fe6982c5f53f941669299dbe086c58a75/api/internal/owner/serializers.py#L253-L276

Reply:
Just a note, you can also use `.nullish()` which provides `<type> | null | undefined` incase you can't fully verify if they'll always return a value.